helm-charts/basehub/values.yaml

```diff
@@ -993,109 +993,120 @@ jupyterhub:
-                # casefold them so we can do case insensitive comparisons, as github itself is case insensitive (but preserving)
-                # for orgs and teams
-                allowed_orgs = set([o.casefold() for o in allowed_teams if ':' not in o])
-                allowed_teams = set([t.casefold() for t in allowed_teams if ':' in t])
+              groups |= set([f'{team["organization"]["login"]}:{team["slug"]}'.casefold() for team in auth_state["teams"]])
 
-                if allowed_teams & teams:
-                    print(f"Allowing profile {profile['display_name']} for user {spawner.user.name} based on team membership")
-                    allowed_profiles.append(profile)
-                    continue
+            print(f"User {spawner.user.name} is part of groups {' '.join(groups)}")
 
-                if "token_response" in auth_state:
-                    access_token = auth_state["token_response"]["access_token"]
-                    token_type = auth_state["token_response"]["token_type"]
+            # Filter out profiles with allowed_groups set if the user isn't part of the group
+            allowed_profiles = []
+            for original_profile in original_profile_list:
+                # Make a copy, as we'll be modifying this profile
+                profile = deepcopy(original_profile)
+
+                # Handle `allowed_groups` specified in profile_options
+                if 'profile_options' in profile:
+                  for k, po in profile['profile_options'].items():
+
+                    # If `unlisted_choice` has an `allowed_groups` and the current
+                    # user is not present in any of those teams, we delete the
+                    # `unlisted_choice` config entirely for this option. The user
+                    # will then not be allowed to 'write in' a value.
+                    if 'unlisted_choice' in po:
+                      if 'allowed_groups' in po['unlisted_choice']:
+                        if not (set(po['unlisted_choice']['allowed_groups']) and groups):
+                          del po['unlisted_choice']
+
+                    if 'choices' in po:
+                      new_choices = {}
+                      for k, c in po['choices'].items():
+                        # If `allowed_groups` is not set for a profile option, it is automatically
+                        # allowed for everyone
+                        if 'allowed_groups' not in c:
+                          new_choices[k] = c
+                        # If `allowed_groups` *is* set for a profile option, it is allowed only for
+                        # members of that team.
+                        elif set(c['allowed_groups']) & groups:
+                          new_choices[k] = c
+                      po['choices'] = new_choices
+
+                # casefold teams so we can do case insensitive comparisons, as github itself is case insensitive (but preserving)
+                # for orgs and teams
```

Comment:
I think this ended up here by accident and should instead be before line 1035 maybe?
```suggestion
```

Reply:
Good catch! Moved!